workshops/boost-research-reproducibility-binder/workshop-presentations/zero-to-binder.md

```diff
@@ -0,0 +1,254 @@
+   **Note the two underscores either side of `version`!**
+
+3) Run the cell to see the version number and a random number printed out
+   * Press either SHIFT+RETURN or the "Run" button in the Menu bar
+
+## 6. Sharing your Work
+
+Binder is all about sharing your work easily and there are two ways to do it:
+* Share the **https://mybinder.org/v2/gh/your-username/my-first-binder/master** URL directly
+
+* Visit **https://mybinder.org**, type in the URL of your repo and copy the Markdown or ReStructured Text snippet into your `README.md` file.
+  This snippet will render a badge that people can click, which looks like this: ![Binder](https://mybinder.org/badge_logo.svg)
+
+**TO DO:** :vertical_traffic_light:
+
+1) Add the **Markdown** snippet from **https://mybinder.org** to the `README.md` file in your repo
+
+   * The grey bar displaying a binder badge will unfold to reveal the snippets.
+    Click the clipboard icon next to the box marked with "m" to automatically copy the Markdown snippet.
+
+2) Click the badge to make sure it works!
+
+## 7. Accessing data in your Binder
+
+Another kind of dependency for projects is **data**.
+There are different ways to make data available in your Binder depending on the size of your data and your preferences for sharing it.
+
+#### Small public files
+
+The simplest approach for small, public data files is to add them directly into your GitHub repository.
+They are then directly encapsulated into the environment and versioned along with your code.
+
+This is ideal for files up to **10MB**.
+
+#### Medium public files
+
+To access medium files **from a few 10s MB up to a few hundred MB**, you can add a file called `postBuild` to your repo.
+A `postBuild` file is a shell script that is executed as part of the image construction and is only executed once when a new image is built, not every time the Binder is launched.
+
+#### Large public files
+
+It is not practical to place large files in your GitHub repo or include them directly in the image that Binder builds.
+The best option for large files is to use a library specific to the data format to stream the data as you're using it or to download it on demand as part of your code.
+
+For security reasons, the outgoing traffic of your Binder is restricted to HTTP or GitHub connections only. You will not be able to use FTP sites to fetch data on mybinder.org.
+
+#### Private files
+
+There is no way to access files which are not public from mybinder.org.
+You should consider all information in your Binder as public, meaning that:
+* there should be no passwords, tokens, keys etc in your GitHub repo;
+* you should not type passwords into a Binder running on mybinder.org;
+* you should not upload your private SSH key or API token to a running Binder.
+
+In order to support access to private files, you would need to create a local deployment of [BinderHub](https://binderhub.readthedocs.io/en/latest/) where you can decide the security trade offs yourselves.
```

Comment:
"trade offs" -> "trade-offs"

Not necessarily create - it would be great if more institutions had specialised BinderHub deployments that their members could use :) Could rephrase to something like "In order to support access to private files, you would need to use a specialised local deployment of [BinderHub](https://binderhub.readthedocs.io/en/latest/) where the security settings are suitable for the type of data you want to use."

Reply:
What if they ask where they could find such a specialised BinderHub?

Reply:
Come to our amazing workshop next week! ✨ 

On a re-read, I think my point should rather have been that it sounds like setting up a custom BinderHub is something that people attending this talk could easily do as a next step, when actually, it's a bigger job. I like how you've kept the solution short, but to me it implies that you should be the one to set up your local BinderHub for your project, whereas it's probably a job that needs support from IT/RSE groups.

Reply:
Yes, agreed it's a bigger job. But I also feel that the BinderHub's springing up in the wild are from a single person/a few people who are interested, like Andrew! It's a difficult one to pitch 😅I guess the point of us contributing back to binder is to make it easier for anyone to come along and begin to build one even if they don't have the explicit expertise in the beginning. And then IT/RSE groups come in for maintenance and governance maybe? I'll have a meditate on how to phrase it, because I don't want to say "this is IT/RSE only" but you're right in that it is a big job. Maybe something like "they are _usually_ built by IT/RSE groups for reasons pertaining to security, maintenance and governance"?

Reply:
What do you think of this phrasing? [`04c4297`](/alan-turing-institute/the-turing-way/pull/235/commits/04c429734c9b7af364f931b35181e85093581d8d)

Reply:
Yep, that's to the point and covers all the cases :)